### PR TITLE
ignore_time property renamed to min_full_scan_interval

### LINKED ISSUE #3597: Rename the ignore-time Vulnerability Detector setting to min_full_scan_interval - Documentation changes
### Description

As part of the issue https://github.com/wazuh/wazuh/issues/7889, we renamed the `ignore-time` Vulnerability Detector setting to `min_full_scan_interval`. This new setting determined the minimum time after which a full scan of Vulnerability Detector will be performed independently of any database change.

This change should be reflected in the documentation by removing the `ignore-time` setting and adding details for the new `min_full_scan_interval` setting.

### PR DESCRIPTION
**Pull request by MiguelazoDS**
## Description
Replaced all references to ignore_time by min_full_scan_interval. This PR closes #3597
## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar.

### CODE REVIEW

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 1]
```diff
@@ -26,16 +26,17 @@ Scan types
...
-This improves the performance and avoids repeated alerts during a configurable time. We have then two different types of scan:
+In any of these cases, the packages and operating system that have already been scanned will be re-scanned if the vulnerability database receives new CVEs information and :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires.
+This improves the performance and avoids repeated alerts during a configurable time. We have then three different types of scan:
 
-- Full scan: The first time, Vulnerability Detector scans every single package installed. After this, all the available packages are scanned again only when the configured :ref:`ignore_time <vuln_det_ignore_time>` expires.
-- Partial scans: Only new packages are scanned while :ref:`ignore_time <vuln_det_ignore_time>` is still valid.
+- Baseline: This scan type will be triggered after Vulnerability Detector is enabled, Vulnerability Detector performs a full scan considering every single package installed without alerting any of these events.
```
Comment (pereyra-m) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:32]: Maybe, we could:

- Mention that the **baseline** scan only happens the _first time_ 
- Avoid repeating `Vulnerability Detector` in the same sentence
- Mention that the OS is also scanned with the packages

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 2]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+- Full scan: In this scan type, all the available packages are scanned again only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information.
+- Partial scans: Only new packages are scanned.
 
 There are few considerations that arise from this behavior:
 
-- Every full scan generates alerts for all the packages, so the alerts are repeated until they get fixed.
-- The user ca not trigger a full scan manually, the only option is to decrease the :ref:`ignore_time <vuln_det_ignore_time>` setting.
+- Every full scan (not baseline) generates alerts for all the packages, so the alerts are repeated until they get fixed.
```
Comment (pereyra-m) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:38]: In the new design, the full scan after a CVE DB update will not alert the same vulnerabilities again, only the new ones.

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 3]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+- The user can not trigger a full scan manually, the only option is to disable and enable Vulnerability Detector to trigger the baseline scan.
```
Comment (pereyra-m) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:39]: We could mention that this workaround needs a restart after every change in the Vulnerability Detector state.

Comment (pereyra-m) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:39]: You found a typo !

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 4]
```diff
@@ -26,16 +26,17 @@ Scan types
...
 - The partial scans generate alerts for new packages, but they do not delete alerts for removed packages.
```
Comment (pereyra-m) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:40]: The partial scans behavior will be different, but we can update this documentation section later.

**Review by pereyra-m — APPROVED** [review]
LGTM!
The image in **Scan types** still mentions the `ignore_time` and this should be updated later.

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst (hunk 2 above)
Comment (palaciosjeremias) [reply source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:38]: I think adding that full scan and partial scan alerts about new vulnerabilities we can avoid this comment, as "full scan" can be interpreted as the "full scan mechanism"

File: source/user-manual/reference/ossec-conf/vuln-detector.rst [hunk 5]
```diff
@@ -77,12 +77,12 @@ Runs updates and vulnerabilities scans immediately when service is started.
...
-.. _vuln_det_ignore_time:
+.. _vuln_det_min_full_scan_interval:
 
-ignore_time
-^^^^^^^^^^^^
+min_full_scan_interval
+^^^^^^^^^^^^^^^^^^^^^^^
 
-Time during which vulnerabilities that have already been alerted will be ignored. When this time hasn't passed yet, only :ref:`partial scans <vuln_det_scan_types>` will be performed.
+Time during which a full scan will not be performed even if the CVEs database is updated. When this time has passed, only a :ref:`full scan <vuln_det_scan_types>` will be performed if the CVEs database has changed.
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply source/user-manual/reference/ossec-conf/vuln-detector.rst:85]: Or maybe we can express it like:
"When the database is updated, the full scan will wait until this time has expired to lunch."

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst (hunk 1 above)
Comment (MiguelazoDS) [reply source/user-manual/capabilities/vulnerability-detection/how_it_works.rst]: Fixed

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst (hunk 2 above)
Comment (MiguelazoDS) [reply source/user-manual/capabilities/vulnerability-detection/how_it_works.rst]: Fixed taking into account both recommendations

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst (hunk 3 above)
Comment (MiguelazoDS) [reply source/user-manual/capabilities/vulnerability-detection/how_it_works.rst]: Fixed

File: source/user-manual/reference/ossec-conf/vuln-detector.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply source/user-manual/reference/ossec-conf/vuln-detector.rst]: I only changed the position of "only", I think it sounds good, but we can discuss if it needs a deeper change.

File: source/deploying-with-puppet/wazuh-puppet-module/reference-wazuh-puppet/wazuh-manager-class.rst [hunk 6]
```diff
@@ -491,8 +491,8 @@ $vulnerability_detector_interval
 
   `Default 5m`
 
-$vulnerability_detector_ignore_time
-  Time during which vulnerabilities that have already been alerted will be ignored.
+$vulnerability_detector_min_full_scan_interval
+  Time interval which a full scan will be triggered if vulnerability database is updated with new CVEs information.
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply source/deploying-with-puppet/wazuh-puppet-module/reference-wazuh-puppet/wazuh-manager-class.rst]: Fixed

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 7]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+This improves the performance and avoids repeated alerts during a configurable time. We have then three different types of scan:
 
-- Full scan: The first time, Vulnerability Detector scans every single package installed. After this, all the available packages are scanned again only when the configured :ref:`ignore_time <vuln_det_ignore_time>` expires.
-- Partial scans: Only new packages are scanned while :ref:`ignore_time <vuln_det_ignore_time>` is still valid.
+- Baseline: This scan type will be triggered after Vulnerability Detector is enabled, Vulnerability Detector performs a full scan considering every single package installed without alerting any of these events.
+- Full scan: In this scan type, all the available packages are scanned again only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information.
+- Partial scans: Only new packages are scanned.
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply source/user-manual/capabilities/vulnerability-detection/how_it_works.rst]: Fixed

File: source/deploying-with-puppet/wazuh-puppet-module/reference-wazuh-puppet/wazuh-manager-class.rst [hunk 8]
```diff
@@ -491,8 +491,8 @@ $vulnerability_detector_interval
 
   `Default 5m`
 
-$vulnerability_detector_ignore_time
-  Time during which vulnerabilities that have already been alerted will be ignored.
+$vulnerability_detector_min_full_scan_interval
+  Time interval in which a full scan will be triggered if vulnerability database is updated with new CVEs information.
```
Comment (DProvinciani) [inline source/deploying-with-puppet/wazuh-puppet-module/reference-wazuh-puppet/wazuh-manager-class.rst:495]:
```suggestion
  Time interval in which a full scan will be triggered if the database of vulnerabilities is updated with new CVEs information.
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 9]
```diff
@@ -26,16 +26,17 @@ Scan types
 ^^^^^^^^^^
 
 The Vulnerability Detector module can run a scan on startup (:ref:`run_on_start <vuln_det_run_on_start>`) and every certain period of time (:ref:`interval <vuln_det_interval>`).
-In any of these cases, the packages that have already been scanned will wait until the :ref:`ignore_time <vuln_det_ignore_time>` expires to be re-scanned.
-This improves the performance and avoids repeated alerts during a configurable time. We have then two different types of scan:
+In any of these cases, the packages and operating system that have already been scanned will be re-scanned if the vulnerability database receives new CVEs information and :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:29]:
```suggestion
In any of these cases, the packages and the operating system that were already scanned will be re-scanned if the database of vulnerabilities receives new CVEs information and :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires.
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 10]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+This improves the performance and avoids repeated alerts during a configurable time. We have then three different types of scan:
 
-- Full scan: The first time, Vulnerability Detector scans every single package installed. After this, all the available packages are scanned again only when the configured :ref:`ignore_time <vuln_det_ignore_time>` expires.
-- Partial scans: Only new packages are scanned while :ref:`ignore_time <vuln_det_ignore_time>` is still valid.
+- Baseline: This scan type will be triggered the first time after Vulnerability Detector is enabled, it performs a full scan considering every single package installed as well as the operating system without alerting any of those events.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:32]:
```suggestion
- Baseline: This scan type will be triggered the first time after Vulnerability Detector is enabled. It performs a full scan for every single package installed as well as the operating system. As a result, the CVEs inventory will have the information of the vulnerabilities detected but no alerts will be generated.
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 11]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+- Full scan: In this scan type, all the available packages are scanned and alerted only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:33]:
```suggestion
- Full scan: In this scan type, every single package installed as well as the operating system are scanned. It runs only when the configured :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` expires and the CVEs database is updated with new information. As a result, the newly detected vulnerabilities and the ones that no longer affect the agent will be alerted.
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 12]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+- Partial scans: Only new packages are scanned and alerted.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:34]:
```suggestion
- Partial scans: Only new packages are scanned and the vulnerabilities detected are alerted.
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 13]
```diff
@@ -26,16 +26,17 @@ Scan types
...
 
 There are few considerations that arise from this behavior:
 
-- Every full scan generates alerts for all the packages, so the alerts are repeated until they get fixed.
-- The user ca not trigger a full scan manually, the only option is to decrease the :ref:`ignore_time <vuln_det_ignore_time>` setting.
+- Full and partial scans will generate alerts for all the new vulnerabilities found after the CVEs database is updated.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:38]: I suggest removing this line. Now, this is explained in the result of each scan type.
```suggestion
```

File: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst [hunk 14]
```diff
@@ -26,16 +26,17 @@ Scan types
...
+- The user can not trigger a full scan manually, the only option is to disable and enable Vulnerability Detector to trigger the baseline scan. For that purpose a service restart is needed.
```
Comment (DProvinciani) [inline source/user-manual/capabilities/vulnerability-detection/how_it_works.rst:39]:
```suggestion
- The user can not trigger a full scan manually. The only option is to disable and enable Vulnerability Detector to trigger the baseline scan. For that purpose, a service restart is needed.
```

File: source/user-manual/reference/ossec-conf/vuln-detector.rst [hunk 15]
```diff
@@ -77,12 +77,12 @@ Runs updates and vulnerabilities scans immediately when service is started.
...
-.. _vuln_det_ignore_time:
+.. _vuln_det_min_full_scan_interval:
 
-ignore_time
-^^^^^^^^^^^^
+min_full_scan_interval
+^^^^^^^^^^^^^^^^^^^^^^^
 
-Time during which vulnerabilities that have already been alerted will be ignored. When this time hasn't passed yet, only :ref:`partial scans <vuln_det_scan_types>` will be performed.
+Time during which a full scan will not be performed even if the CVEs database is updated. When this time has passed, a :ref:`full scan <vuln_det_scan_types>` will be performed only if the CVEs database has changed.
```
Comment (DProvinciani) [inline source/user-manual/reference/ossec-conf/vuln-detector.rst:85]:
```suggestion
The time during which a full scan will not be performed even if the database of vulnerabilities is updated. When this time expires, a :ref:`full scan <vuln_det_scan_types>` will be performed only if the CVEs database has changed.
```

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Review by palaciosjeremias — APPROVED** [review]
LGTM!

**Pull request merged**